npm/src/auth/index.ts

```diff
@@ -46,7 +44,14 @@ export const scheduleTokenRefresh: (
       (expInSecs - (nowInSecs + idleSessionMinValiditySecs)) * 1000
 
     setTimeout(() => {
-      refreshOrReturnToken(keycloak).then(() => scheduleTokenRefresh(keycloak))
+      refreshOrReturnToken(keycloak).then(() => {
```

Comment:
Assuming these extra steps are now necessary to ensure that the upload to s3 continues by updating the signed cookie?

Will this have any impact should this function be used for anything other than the upload. For example keeping the user signed in on the transfer agreement page?

Reply:
It will if we use this `setTimeout` on the other pages but I think we're going to have it so there's a popup saying `Do you want to stay signed in?` and then it runs some javascript when you click ok. We'd have to do a separate function then that calls `refreshOrReturnToken` but doesn't call the cookies API so this function would still just be for the upload page.

Reply:
OK. Worth bearing mind. Thanks